cdap-spark-core-base/src/main/java/co/cask/cdap/app/runtime/spark/SparkRuntimeUtils.java

```diff
@@ -285,6 +294,72 @@ public static void setLocalizedResources(Set<String> localizedResourcesNames,
     return result;
   }
 
+  /**
+   * Initialize a Spark main() method. This is the first method to be called from the main() method of any
+   * spark program.
+   *
+   * @return a {@link Cancellable} for releasing resources.
+   */
+  public static Cancellable initSparkMain() {
+    Thread mainThread = Thread.currentThread();
+    SparkClassLoader sparkClassLoader;
+    try {
+      sparkClassLoader = SparkClassLoader.findFromContext();
+    } catch (IllegalStateException e) {
+      sparkClassLoader = SparkClassLoader.create();
+    }
+
+    final Cancellable unsetClassLoader = SparkRuntimeUtils.setContextClassLoader(sparkClassLoader);
+
+    final SparkExecutionContext sec = sparkClassLoader.getSparkExecutionContext(true);
+    final Service driverService = startDriverServiceIfNecessary(sparkClassLoader.getRuntimeContext(), mainThread);
+    return new Cancellable() {
+      @Override
+      public void cancel() {
+        unsetClassLoader.cancel();
+
+        if (sec instanceof AutoCloseable) {
+          try {
+            ((AutoCloseable) sec).close();
+          } catch (Exception e) {
+            // Just log. It shouldn't throw, and if it does (due to bug), nothing much can be done
+            LOG.warn("Exception raised when calling {}.close()", sec, e);
```

Comment:
hum, actually it doesn't. Will add a toString (the underlying context does. Anything).